src/main/java/com/commercetools/sync/commons/BaseSync.java

```diff
@@ -22,44 +26,54 @@ protected BaseSync(@Nonnull final U statistics, @Nonnull final V syncOptions) {
+     *      is a subclass of {@link BaseSyncStatisticsBuilder} representing the builder of {@code statistics} baked for
+     *      the sync process
      */
-    protected abstract CompletionStage<U> process(@Nonnull final List<T> resourceDrafts);
+    protected abstract CompletionStage<V> process(@Nonnull final List<T> resourceDrafts);
 
 
     /**
      * Given a list of resource (e.g. categories, products, etc..) drafts. This method compares each new resource in
      * this list with it's corresponding old resource in a given CTP project, and in turn it either issues update
      * actions on the existing resource if it exists or create it if it doesn't.
      *
-     * <p>The time before and after the actual sync process starts is recorded in the {@link BaseSyncStatistics}
-     * container so that the total processing time is computed in the statistics.
-     *
      * @param resourceDrafts the list of new resources as drafts.
      * @return an instance of {@link CompletionStage&lt;U&gt;} which contains as a result an instance of {@link U} which
-     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} instance attribute of
-     *      {@link this} {@link BaseSync}.
+     *      is a subclass of {@link BaseSyncStatistics} representing the {@code statistics} of this sync process.
      */
     public CompletionStage<U> sync(@Nonnull final List<T> resourceDrafts) {
-        statistics.startTimer();
-        return process(resourceDrafts).thenApply(resultingStatistics -> {
-            resultingStatistics.calculateProcessingTime();
+        final long startTime = System.currentTimeMillis();
+        return process(resourceDrafts).thenApply(resultingStatisticsBuilder -> {
+            resultingStatisticsBuilder.setProcessingTimeInMillis(System.currentTimeMillis() - startTime);
+            final U resultingStatistics = resultingStatisticsBuilder.build();
+            updateTotalStatistics(resultingStatistics);
             return resultingStatistics;
         });
     }
 
 
     /**
-     * Returns an instance of type U which is a subclass of {@link BaseSyncStatistics} containing all the stats of the
-     * sync process; which includes a report message, the total number of update, created, failed, processed resources
-     * and the processing time of the sync in different time units and in a human readable format.
+     * Returns an instance of type U which is a subclass of {@link BaseSyncStatistics} containing the total stats of
+     * all sync processes performed by {@code this} sync instance.
      *
-     * @return a statistics object for the sync process.
+     * @return a statistics object for all sync processes performed by {@code this} sync instance.
      */
     @Nonnull
     public U getStatistics() {
-        return statistics;
+        synchronized (totalStatisticsLock) {
```

Comment:
why is this block `synchronized` anyway? But if it is why not make the method itself `synchronized`, if the block covers the entire method, instead of having a monitor object/lock?